features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017-2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.deviceconfig.rest.impl;
+
+import java.util.Date;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opennms.core.criteria.CriteriaBuilder;
+import org.opennms.features.deviceconfig.persistence.api.DeviceConfig;
+import org.opennms.features.deviceconfig.persistence.api.DeviceConfigDao;
+import org.opennms.features.deviceconfig.rest.api.DeviceConfigDto;
+import org.opennms.features.deviceconfig.rest.api.DeviceConfigRestService;
+import org.opennms.web.utils.ResponseUtils;
+
+public class DefaultDeviceConfigRestService implements DeviceConfigRestService {
+
+    private final DeviceConfigDao deviceConfigDao;
+
+    public DefaultDeviceConfigRestService(DeviceConfigDao deviceConfigDao) {
+        this.deviceConfigDao = deviceConfigDao;
+    }
+
+    @Override
+    public Response getDeviceConfigs(
+            Integer limit,
+            Integer offset,
+            String orderBy,
+            String order,
+            Integer ipInterfaceId,
+            String deviceType,
+            Long createdAfter,
+            Long createdBefore
+    ) {
+        var criteriaBuilder = new CriteriaBuilder(DeviceConfig.class);
```

Comment:
CriteriaBuilder could be moved a different method

Reply:
I do not understand exactly. Do you mean moving the construction logic of the criteria into a separate method? In that case all arguments would have to be passed to that method. I do not see an advantage in doing so. That logic is used on at one place.

Reply:
It can be used in future in other methods. I'm good with either way